redi-connect-front/src/components/molecules/teaser/SignIn.tsx

```diff
@@ -0,0 +1,18 @@
+import React from "react";
```

Comment:
This is my northern star on colocation: https://kentcdodds.com/blog/colocation

Reply:
Yup, here I was also a bit unsure but I took this path while there will be 2-3 additional teasers and there will be more pages to come. Separating components like this gives you a bit better overview. I would prefer to wait it up until the next teasers come and see if it makes sense to migrate back. 

I think that on the overall structure we should not discuss a lot at this point since this will change down the time, a lot. I'm have to make review of the structure in 1Month and then discuss the overall structure and define updates.

Reply:
Hmmm. While I see your point, I think we'll face this fundamental issue several times over the coming weeks. While I'm open to review it down the line, I think it's worth debating it sooner rather than later, so we land at one generalized principle we can apply everywhere soon.

To frame the issue, I would say the question to ask is _when and how to split components into lower-level molecules instead of co-locating at the page level_.

If we have one component that is inherently part of one page, and that page only (i.e. a 1:1 relationship), I think there is more benefit in co-locating the component inside the file of the page, as compared to separating it out as an atom or molecule. If the file grows very large, we can create a folder containing the page component and its "inherent" components.

In the case of the `SignIn` and `SignUp` teaser components, here is what I have in mind.

1. Create a molecule called `Teaser` (there's some shared code between `SignIn` and `SignUp` that makes this trivial.
2. If we still want to keep the different teasers grouped together, we should re-use a handy pattern we've seen with `react-bulma-components`: `Form.X`. The `Teaser.tsx` file could look like this:
```
function Teaser () {
  // teaser component
}

export default {
  SignIn: () => (
    <Teaser>Content specific to the sign-in teaser</Teaser>
  ),
  SignUp: () => (
    <Teaser>Content specific to the sign-up teaser</Teaser>
  ),
  /* etc */
}
```

... and then it can be used as such:

```
import Teaser from ' .... `

function SomeComponent () {
  return (
    <blabla>
      <Teaser.SignUp />
    </blabla>
  )
}
```